modules/datastore/src/DataDictionary/DateFormat/Converter.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Drupal\datastore\DataDictionary\DateFormat;
+
+use Drupal\datastore\DataDictionary\CompilerInterface;
+use Drupal\datastore\DataDictionary\DateFormatConverterInterface;
+use Drupal\datastore\DataDictionary\ParserInterface;
+
+/**
+ * Converts between date formats.
+ *
+ * The date format parser given is used to parse the input format supplied to
+ * `::convert()`, and the date format compiler is then used to generate the
+ * converted output format.
+ *
+ * The internal language used by ASTs which all parsers are supposed to convert
+ * input format's to is the Frictionless date format.
+ */
+class Converter implements DateFormatConverterInterface {
```

Comment:
When discussing this PR on Slack, @dafeder suggested we break off the date format converter into it's own separate library, since it's a bit esoteric and isolated. I think this is a good idea and certainly worth pursuing.